File: .claude/skills/linkedin-dm/scripts/dm_formatter.py
#!/usr/bin/env python3
"""
linkedin_dm_formatter.py — Formatta, valida e salva DM freddi LinkedIn
Produce 2 varianti: Diretta (profilo business/numeri) e Narrativa (profilo personale/valori)
Uso:
  dm_formatter.py --variant-a "testo diretto" --variant-b "testo narrativo" --lead "Mario Rossi"
  dm_formatter.py --variant-a "testo" --variant-b "testo" --lead "Mario" --copy a
"""

import argparse
import sys
from datetime import date
from pathlib import Path

PLANS_DIR = Path(__file__).parents[4] / "plans"
MIN_WORDS, MAX_WORDS = 150, 180


def validate(text, variant_label):
    words = len(text.split())
    if words < MIN_WORDS:
        print(f"⚠️  Variante {variant_label}: {words} parole — troppo corta (minimo {MIN_WORDS})")
    elif words > MAX_WORDS:
        print(f"⚠️  Variante {variant_label}: {words} parole — troppo lunga (massimo {MAX_WORDS})")
    else:
        print(f"✓ Variante {variant_label}: {words} parole — OK")
    return words


def save(text_a, text_b, lead_name):
    PLANS_DIR.mkdir(parents=True, exist_ok=True)
    safe = lead_name.lower().replace(" ", "_")
    path = PLANS_DIR / f"linkedin_dm_{safe}_{date.today().isoformat()}.txt"
    content = f"=== VARIANTE A — Diretta ===\n\n{text_a}\n\n=== VARIANTE B — Narrativa ===\n\n{text_b}"
    path.write_text(content, encoding="utf-8")
    print(f"✓ Salvato: {path}")
    return str(path)


def copy_to_clipboard(text):
    try:
        import pyperclip
        pyperclip.copy(text)
        print("✓ Copiato negli appunti")
    except ImportError:
        print("⚠️  pyperclip non installato: pip install pyperclip", file=sys.stderr)


def main(text_a, text_b, lead_name, copy_variant=None):
    print(f"\n=== DM LinkedIn — {lead_name} ===")
    print(f"\n--- Variante A (Diretta) ---\n{text_a}")
    print(f"\n--- Variante B (Narrativa) ---\n{text_b}")
    print()
    validate(text_a, "A")
    validate(text_b, "B")
    save(text_a, text_b, lead_name)
    if copy_variant == "a":
        copy_to_clipboard(text_a)
    elif copy_variant == "b":
        copy_to_clipboard(text_b)


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="LinkedIn DM Formatter")
    parser.add_argument("--variant-a", required=True)
    parser.add_argument("--variant-b", required=True)
    parser.add_argument("--lead", required=True)
    parser.add_argument("--copy", choices=["a", "b"])
    args = parser.parse_args()
    main(args.variant_a, args.variant_b, args.lead, copy_variant=args.copy)

File: .claude/skills/linkedin-dm/scripts/dm_formatter.js
#!/usr/bin/env node
/**
 * linkedin_dm_formatter.js — Formatta, valida e salva DM freddi LinkedIn (2 varianti)
 * Uso:
 *   node dm_formatter.js --variant-a "testo diretto" --variant-b "testo narrativo" --lead "Mario Rossi"
 *   node dm_formatter.js --variant-a "testo" --variant-b "testo" --lead "Mario" --copy a
 */

import { mkdirSync, writeFileSync } from 'fs';
import { resolve, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const PLANS_DIR = resolve(__dirname, '../../../../plans');
const MIN_WORDS = 150;
const MAX_WORDS = 180;

function validate(text, variantLabel) {
  const words = text.trim().split(/\s+/).length;
  if (words < MIN_WORDS) console.log(`⚠️  Variante ${variantLabel}: ${words} parole — troppo corta (minimo ${MIN_WORDS})`);
  else if (words > MAX_WORDS) console.log(`⚠️  Variante ${variantLabel}: ${words} parole — troppo lunga (massimo ${MAX_WORDS})`);
  else console.log(`✓ Variante ${variantLabel}: ${words} parole — OK`);
  return words;
}

function save(textA, textB, leadName) {
  mkdirSync(PLANS_DIR, { recursive: true });
  const safe = leadName.toLowerCase().replace(/\s+/g, '_');
  const today = new Date().toISOString().split('T')[0];
  const path = resolve(PLANS_DIR, `linkedin_dm_${safe}_${today}.txt`);
  const content = `=== VARIANTE A — Diretta ===\n\n${textA}\n\n=== VARIANTE B — Narrativa ===\n\n${textB}`;
  writeFileSync(path, content, 'utf8');
  console.log(`✓ Salvato: ${path}`);
  return path;
}

async function copyToClipboard(text) {
  try {
    const { default: clipboardy } = await import('clipboardy');
    clipboardy.writeSync(text);
    console.log('✓ Copiato negli appunti');
  } catch {
    console.error('⚠️  Impossibile copiare negli appunti');
  }
}

async function main(textA, textB, leadName, copyVariant = null) {
  console.log(`\n=== DM LinkedIn — ${leadName} ===`);
  console.log(`\n--- Variante A (Diretta) ---\n${textA}`);
  console.log(`\n--- Variante B (Narrativa) ---\n${textB}\n`);
  validate(textA, 'A');
  validate(textB, 'B');
  save(textA, textB, leadName);
  if (copyVariant === 'a') await copyToClipboard(textA);
  else if (copyVariant === 'b') await copyToClipboard(textB);
}

// CLI
const args = process.argv.slice(2);
const get = (flag) => { const i = args.indexOf(flag); return i !== -1 ? args[i + 1] : null; };

const variantA = get('--variant-a');
const variantB = get('--variant-b');
const lead = get('--lead');

if (!variantA || !variantB || !lead) {
  console.log('Uso: node dm_formatter.js --variant-a "testo" --variant-b "testo" --lead "Nome" [--copy a|b]');
  process.exit(1);
}

main(variantA, variantB, lead, get('--copy')).catch(e => { console.error(e.message); process.exit(1); });
